### PR TITLE
feat: add VictoriaMetrics addon

### PR DESCRIPTION
**Pull request by ButterBright**
- https://github.com/apecloud/kubeblocks-addons/issues/600

### CODE REVIEW

File: addons/victoria-metrics/templates/clusterdefinition.yaml [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+            resources:
+              {}        
+
+    - name: vmselect
+      characterType: vmselect
+      workloadType: Stateless
+      service:
+        ports:
+          - name: http
+            port: 8481
```
Comment (sophon-zt) [inline addons/victoria-metrics/templates/clusterdefinition.yaml:132]: Can you add a parameter to determine whether to enable the ipv6？ `-enableTCP6`

File: addons/victoria-metrics/templates/clusterdefinition.yaml [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+            image: "victoriametrics/vmstorage:v1.100.1-cluster"
+            imagePullPolicy: "IfNotPresent"
+            securityContext:
+              {}
+            args:
+              - "--retentionPeriod=1"
+              - "--storageDataPath=/storage"
+              - --envflag.enable=true
+              - --envflag.prefix=VM_
+              - --loggerFormat=json
```
Comment (sophon-zt) [inline addons/victoria-metrics/templates/clusterdefinition.yaml:205]: Can you add a parameter to determine whether to enable the ipv6？ `-enableTCP6`

File: addons/victoria-metrics-cluster/values.schema.json [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+      "type": "string",
+      "default": "Delete",
+      "enum": [
+        "Delete",
+        "DoNotTerminate",
+        "Halt",
+        "WipeOut"
+      ]
+    },
+    "replicas": {
```
Comment (shanshanying) [inline addons/victoria-metrics-cluster/values.schema.json:35]: update replicas, cpu, memory per component w.r.t  values.yaml
e.g.  replicas -> vminsert.replicas

File: addons/victoria-metrics-cluster/values.yaml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+## @param vmselect configs
+##
+vmselect:
+  replicas: 1
+  resources: {}
+
+## @param vmstorage configs
+##
+vmstorage:
```
Comment (shanshanying) [inline addons/victoria-metrics-cluster/values.yaml:25]: add an extra field,  such that users can specify their storage class name.
e.g. refer to `storageClassName` in apecloud-mysql.

File: addons/victoria-metrics/Chart.yaml [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+description: "VictoriaMetrics is a fast, cost-effective and scalable monitoring solution and time series database."
+
+type: application
+
+version: 0.1.0
+
+appVersion: "v1.100.1"
+
+annotations:
+  addon.kubeblocks.io/kubeblocks-version: ">=0.9.0"
```
Comment (shanshanying) [inline addons/victoria-metrics/Chart.yaml:12]: seems this addon is built on KB 08x api. Maybe you can use >=0.8.0 instead.

File: addons/victoria-metrics-cluster/values.schema.json [hunk 6]
```diff
@@ -32,15 +32,55 @@
         "WipeOut"
       ]
     },
-    "replicas": {
+    "vminsert.replicas": {
```
Comment (shanshanying) [inline addons/victoria-metrics-cluster/values.schema.json:35]: here is an example of schme json.
https://www.arthurkoziel.com/validate-helm-chart-values-with-json-schemas/

Please refine the schme accordingly and make sure it works as expected. ( validate replicas/cpu/memory/ for  each component)

**Review by sophon-zt — APPROVED** [review]

**Review by shanshanying — APPROVED** [review]

**Pull request merged**